test/EVMHost.cpp

```diff
@@ -187,6 +187,15 @@ void EVMHost::recordCalls(evmc_message const& _message) noexcept
 evmc::result EVMHost::call(evmc_message const& _message) noexcept
 {
 	recordCalls(_message);
+
+	// Force failure for Byzantium precompiles on pre-Byzantium chains.
+	if (
+		_message.destination >= 0x0000000000000000000000000000000000000005_address &&
+		_message.destination <= 0x0000000000000000000000000000000000000008_address &&
+		m_evmVersion < langutil::EVMVersion::byzantium()
+	)
+		return evmc::result(EVMC_FAILURE, 0, nullptr, 0);
```

Comment:
Why should pre compile address fail for pre-Byzantium chains? Shouldn't they be treated as an empty account?

Reply:
Yeah I realised that after I pushed this (it was an old PR), but forgot to comment. We could just set it to `SUCESS`. The reason it may make sense merging it that way so it is clear what is going on?